plugins/modules/ipadnsconfig.py

```diff
@@ -0,0 +1,261 @@
+            # shouldn't happen, but let's be paranoid.
+            module.fail(msg="Invalid state: %s" % state)
+
+    if forward_policy is not None:
+        _args['idnsforwardpolicy'] = forward_policy
+
+    if allow_sync_ptr is not None:
+        _args['idnsallowsyncptr'] = 'TRUE' if allow_sync_ptr else 'FALSE'
+
+    return _args
+
+
+def check_parameters(module):
+    pass
+
+
+def main():
+    forwarder_spec = dict(
+       ip_address=dict(type=str, required=True),
+       port=dict(type=int, required=False, default=None)
+    )
+
+    ansible_module = AnsibleModule(
+       argument_spec=dict(
+           # general
+           ipaadmin_principal=dict(type='str', default='admin'),
+           ipaadmin_password=dict(type='str', no_log=True),
+
+           # dnsconfig
+           forwarders=dict(type='list', default=None, required=False,
+                           options=dict(**forwarder_spec)),
+           forward_policy=dict(type='str', required=False, default=None,
+                               choices=['only', 'first', 'none']),
+           allow_sync_ptr=dict(type='bool', required=False, default=None),
+
+           # general
+           state=dict(type="str", default="present",
+                      choices=["present", "absent"]),
+
+       )
+    )
+
+    ansible_module._ansible_debug = True
+
+    # general
+    ipaadmin_principal = module_params_get(ansible_module,
+                                           "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module,
+                                          "ipaadmin_password")
+
+    forwarders = module_params_get(ansible_module, 'forwarders') or []
+    forward_policy = module_params_get(ansible_module, 'forward_policy')
+    allow_sync_ptr = module_params_get(ansible_module, 'allow_sync_ptr')
+
+    state = module_params_get(ansible_module, 'state')
+
+    # Check parameters.
+    invalid = []
+    if state == 'absent':
+        invalid = ['forward_policy', 'allow_sync_ptr']
+
+    for x in invalid:
+        if vars()[x] is not None:
+            ansible_module.fail_json(
+                msg="Argument '%s' can not be used with state '%s'" %
+                (x, state))
+
+    # Init
+
+    changed = False
+    ccache_dir = None
+    ccache_name = None
+    try:
+        if not valid_creds(ansible_module, ipaadmin_principal):
+            ccache_dir, ccache_name = temp_kinit(ipaadmin_principal,
+                                                 ipaadmin_password)
+        api_connect()
+
+        res_find = find_dnsconfig(ansible_module)
+        args = gen_args(ansible_module, state, res_find, forwarders,
+                        forward_policy, allow_sync_ptr)
+
+        # Execute command only if configuration changes.
+        if not compare_args_ipa(ansible_module, args, res_find):
+            try:
+                api_command_no_name(ansible_module, 'dnsconfig_mod', args)
+                # If command did not fail, something changed.
```

Comment:
If this should be set in case the command not failed shouldn't be set in the else block? try/except/else?

Reply:
It could, it shouldn't.